### PR TITLE
Kafka avro encoder

### PR DESCRIPTION
**Pull request by charlesjmorgan**
Add `AvroRowEncoder` and `AvroRowEncoderFactory`
Add test case in `io.prestosql.plugin.kafka.TestKafkaIntegrationSmokeTest#testRoundTripAllFormats`

blocked by [https://github.com/prestosql/presto/pull/4287](url)
@losipiuk @aalbu

### CODE REVIEW

**Comment by losipiuk** [comment]
You can change the base of this PR now to prestosql/presto:master

File: presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/avro/AvroRowEncoder.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    {
+        super(session, columnHandles);
+        this.schema = requireNonNull(parsedSchema, "parsedSchema is null");
+        this.record = new GenericData.Record(parsedSchema);
+    }
+
+    @Override
+    protected void validateColumns(List<EncoderColumnHandle> columnHandles)
+    {
+        for (EncoderColumnHandle columnHandle : columnHandles) {
```
Comment (losipiuk) [inline presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/avro/AvroRowEncoder.java]: As discussed elswhere let's move it to constructor.

File: presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/avro/AvroRowEncoder.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    }
+
+    @Override
+    protected void appendString(String value)
+    {
+        record.put(columnHandles.get(currentColumnIndex).getName(), value);
+    }
+
+    @Override
+    protected void appendByteBuffer(ByteBuffer value)
```
Comment (losipiuk) [inline presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/avro/AvroRowEncoder.java]: It does not seem to be used by any of supported types.

File: presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/avro/AvroRowEncoder.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        record.put(columnHandles.get(currentColumnIndex).getName(), value);
+    }
+
+    @Override
+    public byte[] toByteArray()
+    {
+        // make sure entire row has been updated with new values
+        checkArgument(currentColumnIndex == columnHandles.size(), format("Missing %d columns", columnHandles.size() - currentColumnIndex + 1));
+
+        try (ByteArrayOutputStream byteArrayOuts = new ByteArrayOutputStream();
```
Comment (losipiuk) [inline presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/avro/AvroRowEncoder.java]: This seems quite expensive for being called for each row. 
What about changing it such that `ByteArrayOutputStream bos` and `DataFileWriter dataFileWriter` are fields isn `AvroRowEncoder`.
Then in toByteArray you can just call:
```java
    bos.reset()
    dataFileWriter.append(record);
    dataFileWriter.flush();
    return bos.toByteArray();
```
 
Yet to make this change we will need to make `RowEncoder` closable, and close `ByteArrayOutputStream` and `DataFileWriter` in `AvroRowEncoder.close()`

**Comment by charlesjmorgan** [comment]
have to open a new pr to merge into `presto/master`

**Pull request closed without merging**